src/e2e/java/teammates/e2e/pageobjects/InstructorAuditLogsPage.java

```diff
@@ -0,0 +1,126 @@
+package teammates.e2e.pageobjects;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+/**
+ * Represents the instructor audit logs page of the website.
+ */
+public class InstructorAuditLogsPage extends AppPage {
+    private List<WebElement> sessionCards;
+
+    @FindBy(id = "course-id-dropdown")
+    private WebElement courseIdDropDown;
+
+    @FindBy(id = "student-name-dropdown")
+    private WebElement studentNameDropDown;
+
+    @FindBy(id = "logs-from-datepicker")
+    private WebElement logsFromDatepicker;
+
+    @FindBy(id = "logs-to-datepicker")
+    private WebElement logsToDatepicker;
+
+    @FindBy(id = "logs-from-timepicker")
+    private WebElement logsFromTimepicker;
+
+    @FindBy(id = "logs-to-timepicker")
+    private WebElement logsToTimepicker;
+
+    @FindBy(id = "search-button")
+    private WebElement searchButton;
+
+    @FindBy(id = "logs-output")
+    private WebElement logsOutput;
+
+    public InstructorAuditLogsPage(Browser browser) {
+        super(browser);
+    }
+
+    @Override
+    protected boolean containsExpectedPageContents() {
+        return getPageTitle().contains("Student Activity Logs");
+    }
+
+    public void startSearching() {
+        click(searchButton);
+        sessionCards = logsOutput.findElements(By.className("card"));
+    }
+
+    public List<String> getLogsData() {
```

Comment:
I'm assuming this method retrieves all the logs that are being displayed? In that case, this will never collect the data if there were logs present since you find by `tagname "i"`. The only possibility of that is when there is no logs